R/connect.R

```diff
@@ -20,20 +20,64 @@
 #' instance
 #' }
 connect <- function(slug = NULL) {
-  user_settings <- .settings_load__load_or_create_user_settings()
+  instance_settings <-
+    if (is.null(slug)) {
+      # if the slug is null, see if we can load the default instance
+      instance_file <- .settings_store__current_instance_settings_file()
 
-  if (is.null(slug)) {
-    current_instance <- .settings_load__load_instance_settings()
-    slug <- paste0(current_instance$owner, "/", current_instance$name)
-  }
+      if (file.exists(instance_file)) {
+        .settings_load__load_instance_settings(instance_file)
+      } else {
+        cli_abort(
+          paste0(
+            "Could not load default instance. Either:\n",
+            " - Provide a slug. For example: `connect(\"laminlabs/cellxgene\")`)\n",
+            " - Set a default instance by running `lamin load <slug>`."
+          )
+        )
+      }
+    } else {
+      # if the slug is not null, try to load the instance from the local settings
+      owner_name <- .connect_get_owner_name_from_identifier(slug)
 
-  owner_name <- .connect_get_owner_name_from_identifier(slug)
+      instance_file <- .settings_store__instance_settings_file(
+        name = owner_name$name,
+        owner = owner_name$owner
+      )
 
-  instance_settings <- .connect_get_instance_settings(
-    owner = owner_name$owner,
-    name = owner_name$name,
-    access_token = user_settings$access_token
-  )
+      if (file.exists(instance_file)) {
+        .settings_load__load_instance_settings(instance_file)
+      } else {
+        # try to load the user settings from the api
+        user_file <- .settings_store__current_user_settings_file()
+
+        if (!file.exists(user_file)) {
+          cli_abort(paste0(
+            "No default user or instance is loaded! Either:\n",
+            " - Call `lamin login` to set a default user.\n",
+            " - Call `lamin load <slug>` to set a default instance.\n"
+          ))
+        } else {
```

Comment:
Could this be combined with the similar chunk above? Possibly the logic is easier this way though.

Reply:
Yes -- and indeed!